content/api/commands/origin.md

```diff
@@ -0,0 +1,512 @@
+test, like the following.
+
+<Badge type="success">After</Badge> One big test using `cy.origin()`
+
+```js
+it('securely edits content', () => {
+  cy.origin('supersecurelogons.com', () => {
+    cy.visit('https"//supersecurelogons.com')
+    cy.get('input#password').type('Password123!')
+    cy.get('button#submit').click()
+  })
+
+  cy.origin('mycms.com', () => {
+    cy.url().should('contain', 'cms')
+    cy.get('#current-user').contains('logged in')
+    cy.get('button#edit-1').click()
+    cy.get('input#title').type('Updated title')
+    cy.get('button#submit').click()
+    cy.get('.toast').type('Changes saved!')
+  })
+
+  cy.visit('/items/1')
+  cy.get('h1').contains('Updated title')
+})
+```
+
+Always remember,
+[Cypress tests are not unit tests](https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion).
+
+### Serialization
+
+When entering a `cy.origin()` block, Cypress injects itself at runtime, with all
+your configuration settings, into the requested origin, and sets up
+bidirectional communication with that instance. This coordination model requires
+that any data sent from one instance to another be
+[serialized](https://developer.mozilla.org/en-US/docs/Glossary/Serialization)
+for transmission. It is very important to understand that variables **inside**
+the callback are not shared with the scope **outside** the callback. For example
+this will not work:
+
+**<Icon name="exclamation-triangle" color="red"></Icon> Incorrect Usage**
+
+```js
+const foo = 1
+cy.origin('somesite.com', () => {
+  cy.visit('/')
+  // This line will throw a ReferenceError because
+  // `foo` is not defined in the scope of the callback
+  cy.get('input').type(foo)
+})
+```
+
+Instead, the variable must be explicitly passed into the callback using the
+`args` option:
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```js
+const foo = 1
+cy.origin('somesite.com', { args: { foo } }, ({ foo }) => {
+  cy.visit('/')
+  // Now it will pass
+  cy.get('input').type(foo)
+})
+```
+
+Cypress uses
+[the structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)
+to transfer the `args` option to the secondary origin. This introduces a number
+of
+[restrictions on the data which may be passed](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#things_that_dont_work_with_structured_clone).
+
+### Callback restrictions
+
+Because of the way in which the callback is transmitted and executed, there are
+certain limitations on what code may be run inside it. In particular, the
+following Cypress commands will throw errors if used in the callback:
+
+- `cy.origin()`
+- [`cy.intercept()`](/api/commands/intercept)
+- [`cy.session()`](/api/commands/session)
+- [`cy.server()`](/api/commands/server)
+- [`cy.route()`](/api/commands/route)
+- [`Cypress.Cookies.preserveOnce()`](/api/cypress-api/cookies)
+
```

Comment:
Missing a couple: https://github.com/cypress-io/cypress/blob/feature-multidomain/packages/driver/src/multi-domain/unsupported_apis.ts
Cypress.Server.get()
Cypress.Server.set()
Cypress.session.get()
Cypress.session.set()

Reply:
`get` and `set` aren't the methods. That's us using setters and getters to catch any possible use of `Cypress.Server` and `Cypress.session`.

The user-facing methods are:

- Cypress.Server.defaults()
- Cypress.session.clearAllSavedSessions()